### PR TITLE
[vsphere driver] not working with Esxi free license documentation update

### PR DESCRIPTION
**Pull request by cdosso**
[vsphere driver] not working with Esxi free license

### CODE REVIEW

**Comment by nathanleclaire** [comment]
LGTM cc @ehazlett @moxiegirl

File: docs/index.md [hunk 1]
```diff
@@ -1584,7 +1584,7 @@ Environment variables and default values:
 | `--vmwarevcloudair-docker-port`   | `VCLOUDAIR_DOCKER_PORT`   | `2376`                                     |
 
 #### VMware vSphere
-Creates machines on a [VMware vSphere](http://www.vmware.com/products/vsphere) Virtual Infrastructure. Requires a working vSphere (ESXi and optionally vCenter) installation. The vSphere driver depends on [`govc`](https://github.com/vmware/govmomi/tree/master/govc) (must be in path) and has been tested with [vmware/govmomi@`c848630`](https://github.com/vmware/govmomi/commit/c8486300bfe19427e4f3226e3b3eac067717ef17).
+Creates machines on a [VMware vSphere](http://www.vmware.com/products/vsphere) Virtual Infrastructure. Requires a working vSphere (ESXi with non free license or 60 days trial and optionally vCenter) installation. The vSphere driver depends on [`govc`](https://github.com/vmware/govmomi/tree/master/govc) (must be in path) and has been tested with [vmware/govmomi@`c848630`](https://github.com/vmware/govmomi/commit/c8486300bfe19427e4f3226e3b3eac067717ef17).
```
Comment (moxiegirl) [inline docs/index.md]: Hi @cdosso thank you for the contribution.  You have a lot packed into that sentence.  If we unpack it, hopefully we can clarify.  In researching, I referenced http://www.mustbegeek.com/difference-between-vsphere-esxi-and-vcenter/

**Suggested alternate:**

The machine must have a working vSphere ESXi  installation.  You can use a paid license or free 60 day trial license. Your installation may also include an optional VCenter server.

Comment (cdosso) [inline docs/index.md]: Perfect

**Comment by moxiegirl** [comment]
LGTM

**Comment by ehazlett** [comment]
@frapposelli is this the case?

**Comment by frapposelli** [comment]
@ehazlett shouldn't be the case, but I'll double check.

**Comment by StefanScherer** [comment]
@ehazlett @frapposelli I can confirm this. The vSphere API is not available after inserting the free ESXi license. The API is available running the trial for 60 days without inserting the ESXi license.

It seems that the free ESXi could only be remote controlled [through SSH as packer](https://github.com/mitchellh/packer/blob/master/builder/vmware/iso/driver_esx5.go) does.
Tested this in a [nested environment](https://github.com/StefanScherer/esxi-playground) on my Mac ;-)

**Pull request merged**